Make iSCSI and Fibre Channel grains optional

twangboy opened the pull request:
### What does this PR do?
Creates `iscsi.py` and `fibre_channel.py`
Removes those grains from `core.py`
Gates the grains with options from the config file:
- `fibre_channel_grains`
- `iscsi_grains`

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/46425
https://github.com/saltstack/salt/pull/43647

### Previous Behavior
Grains would (attempt) to be populated every time. It is especially slow on Windows because it uses PowerShell and WMIC.

### New Behavior
These grains are now optional via config

### Tests written?
No

### Commits signed with GPG?
Yes

twangboy committed:
Make iSCSI and Fibre Channel grains optional

Creates iscsi.py and fibre_channel.py
Gates the grains with options from the config file:
- fibre_channel_grains
- iscsi_grains

twangboy committed:
Fix failing tests, add new tests

twangboy committed:
Fix some lint

rallytime commented:
@twangboy This is causing the following test to fail:

- unit.grains.test_iscsi.IscsiGrainsTestCase.test_windows_iscsi_iqn_grains

https://jenkins.saltstack.com/job/PR/job/salt-pr-linode-ubuntu14-n/20844/#showFailuresLink

twangboy committed:
Fix 2 tests

Modifies the fibre_channel.py grain for linux to allow mocking. You
can't mock a file handle used in a for loop.
Fixes the mocking in test_windows_iscsi_iqn_grains. Apparently, you
can't stack multiple mocks in the same with statement.

twangboy commented:
@rallytime I think this is fixed now.

rallytime commented on salt/grains/iscsi.py:
```diff
@@ -0,0 +1,115 @@
+import salt.utils.path
+import salt.utils.platform
+
+__proxyenabled__ = ['iscsi']
```
Are these grains for proxy minions?

twangboy replied:
I don't know, I was just copying other tests. I thought that was for alternate names or something.

rallytime commented on salt/grains/iscsi.py:
```diff
@@ -0,0 +1,115 @@
+
+
+def __virtual__():
+    if __opts__.get('iscsi_grains', False) is False:
```
Is this setting documented anywhere?

twangboy replied:
No

twangboy replied:
It is now

rallytime commented on salt/grains/fibre_channel.py:
```diff
@@ -0,0 +1,77 @@
+import salt.utils.platform
+import salt.utils.files
+
+__proxyenabled__ = ['fibre_channel']
```
Are these grains for proxy minions?

twangboy replied:
See above

rallytime commented on salt/grains/fibre_channel.py:
```diff
@@ -0,0 +1,77 @@
+
+
+def __virtual__():
+    if __opts__.get('fibre_channel_grains', False) is False:
```
Is this setting documented anywhere? I know it's documented in this file, but i mean in other places like config options, release notes, etc.

twangboy replied:
No

twangboy replied:
Also documented

rallytime reviewed (changes requested):
I have a couple of small questions/comments. Otherwise, looks good.

twangboy committed:
Remove unicode things

twangboy committed:
Add configuration options to docs and release notes

Remove __proxyenabled__ from the two new grains files.

Commit:
Merge branch '2018.3.0rc1' into optional_grains

Commit:
Small grammar fix to new grains settings

Commit:
Fix typo

Commit:
Merge branch '2018.3.0rc1' into optional_grains

rallytime commented:
re-run py
